modules/imgproc/include/opencv2/imgproc.hpp

```diff
@@ -1985,20 +1985,22 @@ pixel neighborhood. See cornerEigenValsAndVecs .
 @param useHarrisDetector Parameter indicating whether to use a Harris detector (see #cornerHarris)
 or #cornerMinEigenVal.
 @param k Free parameter of the Harris detector.
+@param corners_values Optional vector of detected corens response.
 
 @sa  cornerMinEigenVal, cornerHarris, calcOpticalFlowPyrLK, estimateRigidTransform,
  */
 
 CV_EXPORTS_W void goodFeaturesToTrack( InputArray image, OutputArray corners,
                                      int maxCorners, double qualityLevel, double minDistance,
                                      InputArray mask = noArray(), int blockSize = 3,
-                                     bool useHarrisDetector = false, double k = 0.04 );
+                                     bool useHarrisDetector = false, double k = 0.04,
+                                     OutputArray corners_values = noArray());
```

Comment:
UB if we write at the noArray() while someone is reading?

Reply:
Added check if writing is actuallly requested (seems the object does not allow write access by default anyway).